wgengine/filter/filter.go

```diff
@@ -26,16 +27,16 @@ import (
 // Filter is a stateful packet filter.
 type Filter struct {
 	logf logger.Logf
-	// local is the set of IPs prefixes that we know to be "local" to
-	// this node. All packets coming in over tailscale must have a
-	// destination within local, regardless of the policy filter
+	// localContains tests where an IP is in the set of IPs prefixes that we
+	// know to be "local" to this node. All packets coming in over tailscale
+	// must have a destination within local, regardless of the policy filter
 	// below.
-	local *netipx.IPSet
+	localContains func(netip.Addr) bool
```

Comment:
i'm actually kinda surprised that this check is the one causing issues and not the one in `matches`

Reply:
oh its the logging one, that makes more sense.